### ISSUE TITLE
Openkore dont work in servers that use EAC ( EasyAntiCheat ) as anticheat

### ISSUE THREAD

**Issue by idcbot12** [OpenKore/openkore#2963]
hello there!! 

i'am already bypassing Ragexe.exe idRO with simple method(hexing code)

Search: D0 C9 4A
Replace: D0 99 4A

now you can run ragexe.exe -1rag1 without EAC(EasyAntiCheat)

but i have some problem..

if you do this, you can login and select character,you can use Scanner or like Cheat Engine, You can Attach Kernel Mode or Accsess Process Ragexe.

the problem is stuck after entering map in client, program stoped working(ragexe stoped working).

anyone help me to complete this.. thanks

**Comment by sctnightcore** [comment]
you cannot login if EAC dont send packet to server .

**Comment by idcbot12** [comment]
thanks for reply sir..

but i can.. try my method sir ... i can login select character but when entering map client ragexe alaways crash

maybe you can help me

**Comment by alisonrag** [comment]
eac encrypt packets in map-server, so if you disable the EAC or remove the pointer, ragnarok will fail in send packets or find the function then will crash

**Comment by alisonrag** [comment]
In the video he is not editing the exe, he use a driver and a dll that disable eac detection, also use functions from eac dll.
he create a local server that receive the packets, then use eac dll functions to send data to server

**Comment by idcbot12** [comment]
how you know step by step?

its you are??

could you help me how do thats??

what must i do..? really i have much question..

**Comment by idcbot12** [comment]
@alisonrag

you right .. i try disabled EAC (skip procedure load module EAC) and i can loginselec character dll

second test: i do disabled and manually load EAC After Netredirect.dll Connect, Result is i cant login filed connect to server

third: i do all and active EAC service open kore working for 1 sec(callculating route to cordinate bla bla) after entering map and bump timeout again or crash again

**Comment by lem-aibon** [comment]
try disabled EAC (skip procedure load module EAC) and i can loginselec character dll..

after that using wpe pro find recpacket

**Comment by idcbot12** [comment]
@lem-aibon 

Already done with disable EAC for skipping load module EA anda i do login etc.. (except enetering maps ive got stopped working)

Useless find RecvPacket using WPE at openkore xkore stuck parses 0A7B, 0A7B is a EAC RecvPacket

alisonrag Logic is good but hard work for implementation lol

**Comment by idcbot12** [comment]
i hope can join discord for fast response discuss

**Comment by lem-aibon** [comment]
try using wireshark https://github.com/alxbl/ro-wireshark

**Comment by lem-aibon** [comment]
try these https://botmysg.blogspot.com/2019/01/bot-ragnarok-mysg.html

**Comment by idcbot12** [comment]
@lem-aibon 

password file the second link??

**Comment by idcbot12** [comment]
and how to use lua file in first link??

**Comment by ya4ept** [comment]
@lem-aibon, what is this viral link?

**Comment by moneyisgod** [comment]
I have made this same way (stuck not working, probably same offset) and then i have changed some registers and i logged in. Moved and etc but the problem is the sync

**Comment by idcbot12** [comment]
tell how you do thats

**Comment by moneyisgod** [comment]
I used some ring0 injection cheat. As i remember i stopped at the same offset. Then i debugged how to bypass this error box, and i just remember that i've changed eax for to 1

**Comment by WagnerMoreira** [comment]
So basically there is no way to use Openkore in servers that have EAC?

**Comment by Fadreus** [comment]
> So basically there is no way to use Openkore in servers that have EAC?

There is a way and people are making cash of it.
Especially those SEA servers from Extreme Electronics.

**Comment by mugemuge001** [comment]
how to bypass eac any methods?

**Comment by Nipodemos** [comment]
🙄 
this thread is to discuss and try to find the bypass @mugemuge001 
if this issue is here is because we didn't find out yet

**Comment by boom19** [comment]
any updates regarding EAC? as of this moment?

**Comment by alisonrag** [comment]
> any updates regarding EAC? as of this moment?

nope

**Comment by erlanda92** [comment]
https://www.google.com/url?sa=t&source=web&rct=j&url=https://m.youtube.com/watch%3Fv%3DupSTqKEPi9M&ved=2ahUKEwjsx8TR1vPnAhUw6XMBHeY-CeQQtwIwD3oECAoQAQ&usg=AOvVaw2yf56SNdLTcYOwDHLjEIEs

Here is the one who succeed, i hope someone can figure out what macro that he use

**Comment by sctnightcore** [comment]
> https://www.google.com/url?sa=t&source=web&rct=j&url=https://m.youtube.com/watch%3Fv%3DupSTqKEPi9M&ved=2ahUKEwjsx8TR1vPnAhUw6XMBHeY-CeQQtwIwD3oECAoQAQ&usg=AOvVaw2yf56SNdLTcYOwDHLjEIEs
> 
> Here is the one who succeed, i hope someone can figure out what macro that he use



it's ukore

**Comment by erlanda92** [comment]
anyone any idea about HWID spoofer?

**Comment by Jayjute** [comment]
don't believe for any bypass for this, with the facebook page like EAC bypass or a youtube video username : Traxv, linking to a discord with disabled comments. with payment method like bitcoin. it's a scam, ive been scammed. after paying ive been blocked in facebook page. learn from my mistake and don't attempt to buy any shady service offering an EAC bypass.

**Comment by rorschach0154** [comment]
Hello, Returning player here for iRO-Thor and iRO-Chaos (since i play both worlds LOL). is Openkore still operational for both servers? if yes can i ask for assistance on which pull request can i read about it?  thank you!

**Comment by render12** [comment]
is there any update about it?

**Comment by android3020** [comment]
hey, im not sure if this will help.
i started the game (Ragexe.exe)
after the game was loaded (not logging in yet)
i went to task manager and stopped EAC service,
then i get back to the game.. i can login and play with no issues...

seems EAC's job is only to make protection during startup then add some on packets maybe.. i dunno XD pardon ima such a noob desperate want kore to work again

**Comment by Fadreus** [comment]
@android3020 

You can help by not comment anything.

And the rest of you who want to ask and give useless stuff like _"is it working?" "Teach me plox" "Hey I got this potato, hope it helps" "Hey, this fishy website is a scam. I know it smells fishy but I thought it is dried squid"_, please follow this procedure:

- Please write that in note pad
- Click Save
- Upload that file into recycle bin
- Click empty recycle bin.

**Comment by uplight** [comment]
I think header should change
not "EAC Bypassing Ragnarok"  but "make Openkore work with EAC"

**Comment by sealreturn12** [comment]
Hi  now I can bypass EAC 
Openkore x3 still works? I never use it.

**Comment by render12** [comment]
> Hi now I can bypass EAC
> Openkore x3 still works? I never use it.

really? how can you do it? can you teach us?

**Comment by sealreturn12** [comment]
> > Hi now I can bypass EAC
> > Openkore x3 still works? I never use it.
> 
> really? how can you do it? can you teach us?
what should i do? :c
![Untitled](https://user-images.githubusercontent.com/62323440/79494271-ed942400-804c-11ea-9f25-2164dbc8d357.png)

**Comment by PipeDeveloper** [comment]
How about if someone that buyed before from Ukore can share all files (dlls and a file that is binded to your machine) and try to crack those files? Openkore always has been recognized for been an open source software, powered by all this amazing people.

**Comment by Theomnikiller** [comment]
I happen to have a zip file that contains everything given from uKore purchaser but are you able to identify the file that would be the "binded to your machine"?  They do request this information:
------------------------------------------------------------------------------------------------------------
2. Please send clear Hwid info
Mac address : 
CPU : 
part id : 
Vendor :
Processor : 
Adaptor : 
UniqueId : 
User :
------------------------------------------------------------------------------------------------------------
I know nothing about coding and the such and I can't upload the zip because its too big. If we can pinpoint what files you want I can drop them here from what were given.

**Comment by ya4ept** [comment]
as soon as the bot appears in the public domain, EAC developers will close the security hole and the bot will stop working

**Comment by Hex7i** [comment]
> I think header should change
> not "EAC Bypassing Ragnarok" but "make Openkore work with EAC"

yeah sir same nProtect

**Comment by Hex7i** [comment]
We will must make new fuction for NetRedirect.dll, For EAC

**Comment by LastRoze** [comment]
Just use Diff Patcher to patch then edit the sclientinfo.xml to point to correct IP address that EAC sends, tada you just bypass it.

This is just a guess tho since you can't pretty much bypass using default OK config, what you need to do actually create your own client that matches pretty much the original exe of the game but now you have a clean version of it without EAC binary intact, aka Hexed.

**Comment by jokinvipo** [comment]
Hey Guys,

Im here to try to help finding a solution for this EAC.
If we get a EAC bypass the openkore can work easy with xkore 1 or we need to update anything more in source code?

**Comment by Hex7i** [comment]
> Just use Diff Patcher to patch then edit the sclientinfo.xml to point to correct IP address that EAC sends, tada you just bypass it.
> 
> This is just a guess tho since you can't pretty much bypass using default OK config, what you need to do actually create your own client that matches pretty much the original exe of the game but now you have a clean version of it without EAC binary intact, aka Hexed.

can't coz, tRo sanding any packet via EAC everytime.

**Comment by Hex7i** [comment]
> We will must make new fuction for NetRedirect.dll, for EAC and EAC Open secure vie ip:port 49.229.28.115:10011

New random port 10011 to 10017

**Comment by Shmerrick** [comment]
Here you go! You need to brute force it for access. Or someone from xkore needs to unlock.
https://drive.google.com/drive/folders/18uzi0_XrDFVGMa4Mm1SN4vGQa8LaRKLG

**Comment by idcbot12** [comment]
@Shmerrick 

What's a password? why you sharing file but its protected password and you dont share the password

**Comment by z3niths** [comment]
I thought that people already know how to break it long ago :p 

Solution: XKore 1 + openkore cli without detected
Clues: 
- Compile netredirect.dll, not to directly send a packet from OPK to the server directly but forward via the client to reuse the encryption function of the client to communicate with the server-side. (You can directly send if you can reverse engineer the encryption algorithm & key out from the client and reuse it directly. This way you can do Xkore 0 ) -- at this step you might need to know the client send&receive bufferAddress which you can easily found it with a Packet Logger. So you can forward it thu client via the address. Once the client updated this address also changed, better to use it offset and modify your net redirect to work with it automatically in case you don't want to waste your time to recompile the dll every single update.

- EAC can be blocked by injecting a singed driver that could control the kernel-level (Windows ring 0) and use that process to manage the EAC thread/process after the exe was run.

- EAC heartbeat check can also be handle by packet capturing between EAC server & client and analyze it. or just patch it with the leaked version. The EAC integrity check can also be break which means you can mod the EAC as you wish from it's compiled binary. (it has a writeup online too) 

PS: I didn't touch it for already 3-4 months, my information probably outdated now. But I used to run it smoothly without paying to any hacker

**Comment by jokinvipo** [comment]
> Here you go! You need to brute force it for access. Or someone from xkore needs to unlock.
> https://drive.google.com/drive/folders/18uzi0_XrDFVGMa4Mm1SN4vGQa8LaRKLG

You sended that the same files (DDLS) that UkoreTeam use to bypass EAC.
Thats files that you sended is a real bypass without any monthly fee or its just for read?
Can you provide the rar password?

Thank you

**Comment by Shmerrick** [comment]
@jokinvipo I don't have the password, that's why I said that it needs to be brute forced or someone needs to give out the password. You have paid for this bot. Please share the password.

**Comment by PipeDeveloper** [comment]
@z3niths Have in mind that most of the 95% of the people here just plays and they dont have programming skills to do what you said.

First point is clear, use xcore1 in simple words

Second point on this case would be great if you can teach how to block EAC from your kernel using that driver, because its obvious you need to modify something at windows folder, the question is exactly what.

If you already compiled a moded EAC binary, can you share it?

**Comment by McTrevor069** [comment]
I am still taking a course degree.

Hey, I am happy to be here now! I noticed that some of our fellas are in Transcendence. I will help upon doing the openkore back with the EAC! hehehe it's end of 2020 we can still do it!

**Comment by Afvg91** [comment]
There are some guys selling EAC bypass on YouTube
https://www.youtube.com/watch?v=s9cPF17Kjkc

**Comment by jacopo-lupi** [comment]
> There are some guys selling EAC bypass on YouTube
> https://www.youtube.com/watch?v=s9cPF17Kjkc

need to know if this guys is a scammer or nor

**Comment by alisonrag** [comment]
> > There are some guys selling EAC bypass on YouTube
> > https://www.youtube.com/watch?v=s9cPF17Kjkc
> 
> need to know if this guys is a scammer or nor

this guy its him, when he sended the video here, first time i clicked in the video, it had just 1 view

**Comment by jacopo-lupi** [comment]
> > > There are some guys selling EAC bypass on YouTube
> > > https://www.youtube.com/watch?v=s9cPF17Kjkc
> > 
> > 
> > need to know if this guys is a scammer or nor
> 
> this guy its him, when he sended the video here, first time i clicked in the video, it had just 1 view

dang

**Comment by alisonrag** [comment]
> > > > There are some guys selling EAC bypass on YouTube
> > > > https://www.youtube.com/watch?v=s9cPF17Kjkc
> > > 
> > > 
> > > need to know if this guys is a scammer or nor
> > 
> > 
> > this guy its him, when he sended the video here, first time i clicked in the video, it had just 1 view
> 
> dang. By the way, can we say that Openkore is dead in the way we were used to know it?

unfortunately EAC team is pretty active, i think they monitor this repo, if someone can figure out how the packet encryption works and release it here, its just a question of time to they patch this

**Comment by sctnightcore** [comment]
> > > > There are some guys selling EAC bypass on YouTube
> > > > https://www.youtube.com/watch?v=s9cPF17Kjkc
> > > 
> > > 
> > > need to know if this guys is a scammer or nor
> > 
> > 
> > this guy its him, when he sended the video here, first time i clicked in the video, it had just 1 view
> 
> dang. By the way, can we say that Openkore is dead in the way we were used to know it?

### **Love Never Dies open source as well.** 
**sctnightcore  04/03/2021**

**Comment by DeBrasa** [comment]
> Existem alguns caras vendendo o desvio de EAC no YouTube
> https://www.youtube.com/watch?v=s9cPF17Kjkc

### **This user practices financial scam. I was a victim of it.**

**Comment by alisonrag** [comment]
> > Existem alguns caras vendendo o desvio de EAC no YouTube
> 
> ### **This user practices financial scam. I was a victim of it.**

i feel for your lost, hope u can get your money back someday

-- Off
deleted all links, this can lead to scammers

**Comment by DeBrasa** [comment]
> i feel for your lost, hope u can get your money back someday
> 
> -- Off
> deleted all links, this can lead to scammers

Unfortunately I made an Amazon Gift Card. Scammers ask for payments by crypto and Gift Cards.

**Comment by DeBrasa** [comment]
**### Did we get any solution to make OpenKore work?**

**Comment by alisonrag** [comment]
> > i feel for your lost, hope u can get your money back someday
> > -- Off
> > deleted all links, this can lead to scammers
> 
> Unfortunately I made an Amazon Gift Card. Scammers ask for payments by crypto and Gift Cards.

can u report this to amazon?

**Comment by DeBrasa** [comment]
> > > i feel for your lost, hope u can get your money back someday
> > > -- Off
> > > deleted all links, this can lead to scammers
> > 
> > 
> > Unfortunately I made an Amazon Gift Card. Scammers ask for payments by crypto and Gift Cards.
> 
> can u report this to amazon?

I did not find an option to report sending a gift card. It is a gift.

**Comment by DeBrasa** [comment]
**Any solution to make OpenKore work?**

**Comment by ivanrodrigues95** [comment]
F for bRO openkore 😢

**Comment by Listerinee** [comment]
F

**Comment by zazaass27** [comment]
Ro GGT Classic Bypass Pls 🙏🏼😢

**Comment by faezalhasriq92** [comment]
Any update on this? Openkore still not working for iRO Chaos Server?

**Comment by Nipodemos** [comment]
basically most developers gave up trying to find the solution (at least I don't know anyone that is trying)

**Comment by PipeDeveloper** [comment]
@Nipodemos for people that plays on iRO, you may still see some openkore users such "ukore" that works over xcore1 and some kind of bypass. Obviusly that person sells a "licence" and thats why most of us cannot bot on there.
If we could crack that kind of bypass, it should be a great contribution

**Comment by pumba-dev** [comment]
I am hopeful for a solution.
BRO lives!

**Comment by TaylorAvila** [comment]
no solution EAC?

**Comment by LastRoze** [comment]
Free? No - Paid Solution? Yes

**Comment by TaylorAvila** [comment]
I don't know anyone who is correct I saw many scams what is the average price of this solution?

**Comment by LastRoze** [comment]
From my personal experience, it's around 10± USD per month, not going to tell publicly what service I was using though.

**Comment by TaylorAvila** [comment]
could you send me private taylorcmq@gmail.com

**Comment by TaylorAvila** [comment]
me parece que não é mais possivel desabilitar o EAC mudando o D0 C9 4A bo bRO
